### DISCUSSION TITLE
Code (Javascript/Typescript) Based Migrations

### DISCUSSION THREAD

**Discussion by BallerIndustries** [prisma/prisma#17965] [Q&A]
Sometimes I want to write migrations in Javascript/Typescript instead of SQL. I know SQL migrations are safer, but there are situations where I can't write a SQL migration.

Other packages support this feature:
https://flywaydb.org/documentation/concepts/migrations.html#java-based-migrations

Is this something possible in Prisma?

**Comment by nurul3101** [comment]
Hello @BallerIndustries 👋 

At the moment it's not natively supported, but we do have a feature request here:
- https://github.com/prisma/prisma/issues/4688

Once we implement this, it should be possible to write migrations in javascript/typescript.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
